### PR TITLE
fix-#126-hyperlinks

### PR DESCRIPTION
**Pull request by pbhatnagar3**
Testing
For testing the following worksheet is used

```
// Editing worksheet links(0xe87e3769a23947d38e58bdebbc28c068).
// https://github.com/codalab/codalab-cli/worksheet-syntax.md
//

% schema s2
% add uuid
% add Parameters uuid "key uuid | add text params | add path /output"
% display table s2
[run topological-sort-graph-1 -- sort.py:topological-sort.py,input:graph-1.txt : python sort.py < input > output]{0xdafcc128243d46fb8a7f76926148ed7a}
```
Now a link to files inside the bundles can be created
image

When the param link in the Parameters column is clicked, you get the following:
image

### CODE REVIEW

File: codalab/apps/web/static/js/worksheet/table_bundle_interface.js [hunk 1]
```diff
@@ -220,6 +220,12 @@ var TableBundle = React.createClass({
         });
         var focusIndex = this.state.rowFocusIndex;
         var row_items = item.interpreted[1];
+        var column_with_hyperlinks = [];
+        (Object.keys(row_items[0])).map(function(x){
```
Comment (percyliang) [inline codalab/apps/web/static/js/worksheet/table_bundle_interface.js]: map => forEach

File: codalab/apps/web/static/js/worksheet/table_bundle_interface.js [hunk 2]
```diff
@@ -220,6 +220,12 @@ var TableBundle = React.createClass({
         });
         var focusIndex = this.state.rowFocusIndex;
         var row_items = item.interpreted[1];
+        var column_with_hyperlinks = [];
```
Comment (percyliang) [inline codalab/apps/web/static/js/worksheet/table_bundle_interface.js]: Write comment with an example of what row_items looks like

File: codalab/apps/web/static/js/worksheet/table_bundle_interface.js (hunk 1 above)
Comment (percyliang) [inline codalab/apps/web/static/js/worksheet/table_bundle_interface.js]: nit: add space before { (also everywhere below)

File: codalab/apps/web/static/js/worksheet/table_bundle_interface.js [hunk 3]
```diff
@@ -291,11 +301,18 @@ var TableRow = React.createClass({
                 )
             }
             // if first element we want to link to bundle
-            if(index == 0){
+            if(column_with_hyperlinks.indexOf(header_key) >= 0 || index == 0){
+                if(column_with_hyperlinks.indexOf(header_key) >= 0){
+                    bundle_url = '/api/bundles/filecontent/' + bundle_uuid + row_items[header_key]['path'];
+                    var content = row_items[header_key]['text'];
```
Comment (percyliang) [inline codalab/apps/web/static/js/worksheet/table_bundle_interface.js]: if 'text' doesn't exist, just default to the basename of the path

File: codalab/apps/web/static/js/worksheet/table_bundle_interface.js [hunk 4]
```diff
@@ -291,11 +301,18 @@ var TableRow = React.createClass({
                 )
             }
             // if first element we want to link to bundle
-            if(index == 0){
+            if(column_with_hyperlinks.indexOf(header_key) >= 0 || index == 0){
+                if(column_with_hyperlinks.indexOf(header_key) >= 0){
+                    bundle_url = '/api/bundles/filecontent/' + bundle_uuid + row_items[header_key]['path'];
```
Comment (percyliang) [inline codalab/apps/web/static/js/worksheet/table_bundle_interface.js]: Define local variable to be equal to row_items[header_key] and then use that

**Comment by percyliang** [comment]
Minor comments - LGTM otherwise

File: codalab/apps/web/static/js/worksheet/table_bundle_interface.js (hunk 1 above)
Comment (pbhatnagar3) [inline codalab/apps/web/static/js/worksheet/table_bundle_interface.js]: Professor, are you referring to use-case like``map(function(x){ `` (where parenthesis are followed by curly brackets) or just everywhere we use a {

Comment (percyliang) [inline codalab/apps/web/static/js/worksheet/table_bundle_interface.js]: I just meant `function(x) {` rather than `function(x){`, not in general.

**Pull request merged**